pydantic/types.py

```diff
@@ -1026,39 +1158,86 @@ def __repr__(self) -> str:
 
 
 class EncoderProtocol(Protocol):
+    """Protocol for encoding and decoding data to and from bytes."""
+
     @classmethod
     def decode(cls, data: bytes) -> bytes:
-        """Can throw `PydanticCustomError`."""
+        """Decode the data using the encoder.
+
+        Args:
+            data: The data to decode.
+
+        Returns:
+            The decoded data.
+        """
         ...
 
     @classmethod
     def encode(cls, value: bytes) -> bytes:
+        """Encode the data using the encoder.
+
+        Args:
+            value: The data to encode.
+
+        Returns:
+            The encoded data.
+        """
         ...
 
     @classmethod
     def get_json_format(cls) -> str:
+        """Get the JSON format for the encoded data.
+
+        Returns:
+            The JSON format for the encoded data.
+        """
         ...
 
 
 class Base64Encoder(EncoderProtocol):
+    """Base64 encoder."""
+
     @classmethod
     def decode(cls, data: bytes) -> bytes:
+        """Decode the data from bytes to a base64 encoded bytes.
```

Comment:
From base64 encoded bytes to original bytes. Encoded state is base64, decoded is raw data.

BTW, the `encode` method has a correct doc string.

Reply:
```suggestion
        """Decode the data from base64 encoded bytes to original bytes data.
```